modules/datastore/src/Form/DashboardForm.php

```diff
@@ -537,26 +531,51 @@ protected function buildResourcesRow($dist): array {
     if (is_array($dist) && isset($dist['distribution_uuid'])) {
       $postImportResult = $this->postImportResultFactory->initializeFromDistribution($dist);
       $postImportInfo = $postImportResult->retrieveJobStatus();
-      $status = $postImportInfo ? $postImportInfo['post_import_status'] : "waiting";
+      $postImportStatus = $postImportInfo ? $postImportInfo['post_import_status'] : "waiting";
       $error = $postImportInfo ? $postImportInfo['post_import_error'] : NULL;
+      $mime_type = in_array($dist['mime_type'], DataResource::IMPORTABLE_FILE_TYPES);
 
-      return [
-        [
-          'data' => [
-            '#theme' => 'datastore_dashboard_resource_cell',
-            '#uuid' => $dist['distribution_uuid'],
-            '#file_name' => basename($dist['source_path']),
-            '#file_path' => UrlHostTokenResolver::resolve($dist['source_path']),
-          ],
-        ],
-        $this->buildStatusCell($dist['fetcher_status']),
-        $this->buildStatusCell($dist['importer_status'], $dist['importer_percent_done'], $this->cleanUpError($dist['importer_error'])),
-        $this->buildPostImportStatusCell($status, $error),
-      ];
+      return $this->buildResourceData($dist, $mime_type, $postImportStatus, $error);
     }
+
     return ['', '', '', ''];
   }
 
+  /**
+   * Build resource data array.
+   *
+   * @param array $dist
+   *   Distribution details.
+   * @param bool $mime_type
+   *   Whether the mime type is importable.
+   * @param string $postImportStatus
```

Comment:
```suggestion
   * @param string $post_import_status
```
Use snake_case for consistency